pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -53,35 +53,49 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		return nil
 	}
 
-	// Check if the latest deployment already exists
-	if deployment, err := c.deploymentClient.getDeployment(config.Namespace, deployutil.LatestDeploymentNameForConfig(config)); err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("couldn't get deployment for config %s: %v", deployutil.LabelForDeploymentConfig(config), err)
-		}
-	} else {
-		// If there's an existing deployment, nothing needs to be done.
-		if deployment != nil {
-			return nil
-		}
-	}
-
-	// Check if any previous deployment is still running (any non-terminal state).
+	// Check if any existing inflight deployments (any non-terminal state).
 	existingDeployments, err := c.deploymentClient.listDeploymentsForConfig(config.Namespace, config.Name)
 	if err != nil {
 		return fmt.Errorf("couldn't list deployments for config %s: %v", deployutil.LabelForDeploymentConfig(config), err)
 	}
+	var inflightDeployment, deploymentForCancellation *kapi.ReplicationController
```

Comment:
Are you just suggesting that `deploymentForCancellation` be scoped inside the for loop since its not referred outside it? Sure, I will do that.